### PR TITLE
Added integration for error messages with bots, fixed Start UI button issues

### PR DESCRIPTION
**Pull request by nicorsm**
### Issues

- App was not displaying any error while adding bots from the Start UI
- After closing the Add Bots screen, the Start UI button is disabled until another operation is done (e.g. opening a conversation)

### Solutions

I've handled the errors provided by the Sync Engine (`AddBotResult` and `AddBotError` enums) and, by consequence, I've attached the relative delegate methods.

### CODE REVIEW

**Comment by mikeger** [comment]
@nicorsm sorry, my PR created some conflicts.

**Comment by nicorsm** [comment]
@mikeger I think there's a problem with error handling. While offline, I've received `general` instead of `offline` as `AddBotError` response.

File: Wire-iOS/Sources/UserInterface/Services/ServiceDetailViewController.swift [hunk 1]
```diff
@@ -67,18 +67,23 @@ private func add(service: Service, to conversation: Any) {
...
-        }
+            
+            completion(result)
+        })
     case .existing(let conversation):
-        conversation.add(serviceUser: service.serviceUser, in: userSession) { error in
-            if let _ = error {
-                // TODO: Handle Error
+        conversation.add(serviceUser: service.serviceUser, in: userSession) { done in
+            if let done = done {
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Services/ServiceDetailViewController.swift]: The error name `done` sounds a bit misleading to me, I am not sure if this is a `Bool` indicating whether the operation succeeded or an `Error`, but assume it's the latter?

Comment (nicorsm) [reply Wire-iOS/Sources/UserInterface/Services/ServiceDetailViewController.swift]: @daehn exactly, `done` represents an enum of type `AddBotError`. I'll replace it with `error`.

**Comment by mikeger** [comment]
@nicorsm I'll check the backing logic of the errors and what's the issue.

**Pull request merged**